jax/_src/compiler.py

```diff
@@ -328,9 +328,16 @@ def compile_or_get_cached(
       module_name, cache_key, compile_options, backend)
   cache_retrieval_time = time.monotonic() - cache_retrieval_start
 
+  log_priority = (logging.WARNING if config.explain_cache_misses.value
+                    else logging.DEBUG)
+  cache_hit_msg = "PERSISTENT COMPILATION CACHE HIT for '%s'"
+  log_cache_hit = lambda: logger.log(log_priority, cache_hit_msg, module_name)
```

Comment:
If `explain_cache_misses` is True, you just want to explain the misses (not hits).

And when explaining misses, you probably want to give more information about the cache key so that it's easier to track down what caused the miss.

If you just want to log regardless, I would recommend using `jax_log_compiles` config instead of `explain_cache_misses`.

Reply:
So we actually went back and forth on this... `explain_cache_misses` does literally say it should only explain misses :stuck_out_tongue: 

HOWEVER, for the persistent cache, hits have historically been useful for debugging as well because sometimes there are false cache hits due to a bug in the cache key. Plus this doesn't introduce much extra logging since you only get about one cache hit per computation per process. So the reasoning was better to include useful logging even if it doesn't actually match the config name -- like it logs everything you need to debug all caches.

Thinking about this more though, I think I agree that `explain_cache_misses` should only be for misses, since those are usually user error (although sometimes also cache key bugs) whereas false cache hits are always a jax bug. I like your idea of putting it in `jax_log_compiles`, since there should be an easy way to log cache hits IMO (e.g. for debugging or understanding what executables are being produced).

@rdyro WDYT about putting this message under `jax_log_compiles`?

> And when explaining misses, you probably want to give more information about the cache key so that it's easier to track down what caused the miss.

Agree. This will require more plumbing, and this is meant to be a simple PR :) Just logging the info we have will be more useful than nothing, and it leaves the door open for future improvements.

Reply:
Operating under the constraint of keeping this PR simple, I agree with you @skye . It'd be nice to create a single place in code where a cache miss is explained with a reason going forward. 

Breaking down `explain_cache_misses` we currently have two places in the code that relate to this
* cache is not consulted because of (an undocument, but valid) reason [here](https://github.com/google/jax/blob/c86b5f7281bf559499acc4c250c19581f29586b6/jax/_src/compiler.py#L344)
* cache is not *written* because of a documented reason [here](https://github.com/google/jax/blob/c86b5f7281bf559499acc4c250c19581f29586b6/jax/_src/compilation_cache.py#L199)

For now, for this PR, I can 
* move cache hits under `jax_log_compiles`
* second part:
    * log cache-not-writing explanations and un-commented cache misses when `explain_cache_misses`
    * OR keep cache-not-writing in DEBUG and just log un-commented cache misses when `explain_misses`

@skye @yashk2810 WDYT?

Reply:
I would split the log and explain_cache_misses in 2 PRs. Would be easier to review.